src/type/schema.js

```diff
@@ -234,7 +248,24 @@ export class GraphQLSchema {
     if (isUnionType(abstractType)) {
       return abstractType.getTypes();
     }
-    return this._implementations[abstractType.name];
+
+    const types = this._objectImplementations[abstractType.name];
+    if (!types) {
+      return (this._objectImplementations[abstractType.name] = []);
+    }
+
+    return types;
+  }
+
+  getImplementations(
```

Comment:
This should be a separate PR: this functionality is  useful even without interfaces implementing other interfaces, and should be considered on its own merits. Especially as you're not using it in this PR directly.

Reply:
Thanks so much for the excellent review! I just went to split these out, but their language and types _do_ assume that interfaces are potentially valid implementors. Are these assumptions appropriate to break out into an unrelated? I'm more than happy to do it, but I would be cautions of actually merging them without the rest, as the assumptions are only true in the context of this RFC.

Reply:
@mjmahone just to bump this and make sure I'm clear on the rationale for breaking this into a separate PR: is the thought that this is changing the public API of this type in a way that is useful even without the interface RFC? I am a bit reluctant to do this, because in a two-PR scenario I would introduce a non-breaking change in the first one (adding the methods, which only ever return concrete types simply because those are the only thing that can currently implement an interface); then introduce a potentially-breaking change in the second (these methods would now return both concrete types _and_ interfaces). Introducing these methods in the form written here makes sure there's only one, non-breaking change.

Reply:
@mjmahone @mike-marcacci I don't think it's worth to extract this into separate PR since `getImplementations(interaceType)` is fully queal to `getPossibleTypes(interfaceType)` without this PR being merged.